segments: Add missing API for "dot" control on big 7-segments

vdehors opened the pull request:
There are 4 side LEDs around the 4 7-segments on 1.2" packages of HT16k33
(https://www.adafruit.com/product/1270). This commit adds an API to control
them (set and get).

There already was setter and getter to control "ampm" dot (the one at the
top-right). This API is still working but now use the new added functions.

vdehors committed:
segments: Add missing API for "dot" control on big 7-segments

There are 4 side LEDs around the 4 7-segments on 1.2" packages of HT16k33
(https://www.adafruit.com/product/1270). This commit adds an API to control
them (set and get).

There already was setter and getter to control "ampm" dot (the one at the
top-right). This API is still working but now use the new added functions.

tannewt reviewed (changes requested):
Thank you for this change!

In general we avoid get/set methods because they can be implemented with properties (like `ampm`) or data descriptors like `colon`. ampm and colon already cover two of the four cases you are adding support here.

I'd recommend adding two more properties for top_left_dot and bottom_left_dot that are implemented like ampm. You can use your new get/set methods internally by adding an _ to the start of the name. The other way do do it is by using [a data descriptor](https://docs.python.org/3/howto/descriptor.html) class for all three.

vdehors committed:
segments: Add properties for LED indicators of BigSeg7x4

vdehors commented:
Hi tannewt,

Indeed, I didn't see the Colon class (which implemented what I needed btw). I made your changes but the "colon" property may be enough (with doc) ?

tannewt commented on adafruit_ht16k33/segments.py:
```diff
@@ -292,22 +294,48 @@ def setindicator(self, index, value):
         return self._get_buffer(0x04) & bitmask
 
+    @property
+    def two_dots_center(self):
```
Isn't this the colon? There shouldn't be two ways to set it.

tannewt reviewed (changes requested):
One comment about two_dots_center. Good otherwise. Thanks!

bobdog6861 commented:
I would like to disable the zero blanking of first and second digits, so that I can display the zeros.  I cannot seem to figure out where in the driver it is making that decision.   Is it in _put, in ht16k33.py  ?

tannewt commented:
@bobdog6861 unless you are testing with this version, please open a separate issue for your question.

makermelissa commented:
I'm thinking this PR may not be necessary. I'm going to attempt to set up a small code example to light up all segments and dots and will paste it here if I can get it working.

makermelissa commented:
Ok, here's how you fill each thing with the way the library currently is now:
```
import board
from adafruit_ht16k33 import segments
i2c = board.I2C()
display = segments.BigSeg7x4(i2c)

# Clear the display.
display.fill(0)

# Fill Everything
display.print("8888") # Each Digit. You can also use display[0]
display.colon[0] = True  # Middle Colon
display.colon[1] = True  # Left Colon
display.ampm = True  # Upper Dot
```

makermelissa commented:
@vdehors, I am going to reopen this. I now understand this adds the ability to add separate control of the left side dots, which is excellent. However, as @tannewt suggested, we probably should remove the two_dots_center setter and getters because of the redundancy.

kattni commented:
@makermelissa Thank you for reviewing this PR! Please consider making the necessary changes yourself so we can get this merged. If you need assistance with that, let me know. Thanks!

vdehors committed:
segments: Removing property two_dots_center

The "colon" property already allow to control these LEDs so remove
two_dots_center because of the redundancy.

makermelissa reviewed (approved):
Looks good. Thank you!
